backend/tournesol/views/unconnected_entities.py

```diff
@@ -0,0 +1,81 @@
+"""
+API endpoints to show unconnected entities
+"""
+from collections import defaultdict
+
+from django.db.models import ObjectDoesNotExist
+from django.http import Http404
+from drf_spectacular.utils import extend_schema, extend_schema_view
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
+
+from tournesol.views.mixins.poll import PollScopedViewMixin
+
+from ..models import Comparison, Entity
+from ..serializers.entity import EntitySerializer
+
+
+@extend_schema_view(
+    get=extend_schema(description="Show unconnected entities for the current user")
+)
+class UnconnectedEntitiesView(
+    PollScopedViewMixin,
+    generics.ListAPIView
+):
+    """
+    API view for showing unconnected entities
+    """
+    serializer_class = EntitySerializer
+    permission_classes = [IsAuthenticated]
+
+    __already_visited_node__ = set()
+    __all_entites__ = set()
+    __all_connections__ = defaultdict(set)
+
+    def get_related_entities(self, entity_id):
+
+        self.__already_visited_node__.add(entity_id)
+        self.__all_entites__.remove(entity_id)
+
+        related_entities = set()
+        for other_node in self.__all_connections__.get(entity_id):
+            if other_node not in self.__already_visited_node__:
+                related_entities.update(self.get_related_entities(other_node))
+
+        return self.__all_connections__.get(entity_id).union(related_entities)
+
+    def get_queryset(self):
+        # Get related entities from source
+        source_node = Entity.objects.none()
+
+        try:
+            entity_uid = self.kwargs.get("uid")
+            source_node = Entity.objects.get(uid=entity_uid)
+        except ObjectDoesNotExist:
+            raise Http404
+
+        comparisons = list(Comparison.objects.filter(
+            poll=self.poll_from_url, user=self.request.user
+        ))
+
+        self.__all_entites__ = set(c.entity_1_id for c in comparisons).union(
```

Comment:
It looks like this set is never used, except for when elements are removed from it (line 38). 

One possible refactor would be to completely remove it? 

However I would probably prefer to make use of it. For example `get_related_entities` could not return anything but instead simply empty the `__all_entites__` as it is currently doing (could be renamed in this case `remove_connected_entities`). Then the final `return` statement could be simplified to use `self.__all_entities__`.

Reply:
Yeah thanks a lot, you can see this comment https://github.com/tournesol-app/tournesol/pull/663#discussion_r856302769 with the switch of purpose of this element